### ISSUE TITLE
Question: Calculation of Comment Ratio

### ISSUE THREAD

**Issue by wienans** [priv-kweihmann/multimetric#83]
Hey i tried double check the results of the Tool with radon and stumbled across some huge differences in Comment Ratio.
I checked the source code and found this: https://github.com/priv-kweihmann/multimetric/blob/6853074cc8ed075280ff841ea3952cc767f41b54/multimetric/cls/metric/comments.py#L29-L37

I wanted to ask if my understanding is correct that you calculate the literal character length of the comment in comparison to the literal character length of the program?

**Comment by priv-kweihmann** [comment]
Yes, that is correct - it might be other tools base that on lines or something else, but for me the essence is characters belonging to comments against the characters of the whole file.

The ratio should be correct, but maybe not comparable to other tools.

In a nutshell that applies to all of the metrics, as it should only be used to compare two or more runs of the tool against each other to decide what is okay and what's not

**Comment by wienans** [comment]
Okay thanks for your answer. Yeah also a understandable decision.

Sure I also only expected that it is supposed to compare with itself. But as it calculates the maintainability index and for example the Microsoft one provides thresholds I wanted to roughly check if the calculations of the sub metrics are somewhat overlapping with what I would expect. And as I actually found it a good idea of the SEI index to increase it via comment ratio I bumped into the comments calculation.
